lib/msal-browser/src/telemetry/BrowserPerformanceClient.ts

```diff
@@ -80,25 +90,24 @@ export class BrowserPerformanceClient extends PerformanceClient implements IPerf
             return;
         }
 
-        const preQueueTimesByEvents = this.preQueueTimeByCorrelationId.get(correlationId);
-
-        if (preQueueTimesByEvents){
-            preQueueTimesByEvents.set(eventName, window.performance.now());
-            this.preQueueTimeByCorrelationId.set(correlationId, preQueueTimesByEvents);
-        } else {
-            const preQueueTimes = new Map();
-            preQueueTimes.set(eventName, window.performance.now());
-            this.preQueueTimeByCorrelationId.set(correlationId, preQueueTimes);
+        const preQueueEvent: PreQueueEvent | undefined = this.preQueueTimeByCorrelationId.get(correlationId);
+        /**
+         * Manually complete queue measurement if there is an incomplete pre-queue event.
+         * Incomplete pre-queue events are instrumentation bugs that should be fixed.
```

Comment:
Would be great if you can author a kusto query to call out where these bugs are occurring.  We can then run that periodically or as part of a dashboard to find these issues.

Reply:
Added new `queuedManuallyCompletedCount` telemetry data point to calculate manually completed queue measurements. We won't be able to tell which ones are closed manually by looking at telemetry though, only the total number. To get more details we will have to investigate logs.

Reply:
Will add a Kusto query later on once changes go live.